### DISCUSSION TITLE
Search filters

### DISCUSSION THREAD

**Discussion by NFanoe** [squidfunk/mkdocs-material#7108] [Ideas]
I know that work is ongoing to improve the search. But I'd like to know if this idea is possible in today's version:

- In the search box, add a dropdown-list with filter options taken from the `mkdocs.yml`.
- Perform a search.
- Select a filter option, and the search results list is narrowed.

Use case: Apart from the actual documentation, the doc site contains thousands of API doc pages. These pages prevent you from finding the right doc page, so you'd want to filter them away.

The filter could be based on the location of the file, which is compared with the `location` in the `search_index.json` file. For instance, if the location contains "/openapi/", don't show it unless the user actively opts to view openapi doc files.

Is this possible - is there a "process_search_result" hook or something?

**Comment by squidfunk** [comment]
You can always fork the theme and add any functionality you want ☺️ However, note that my experience with Lunr.js, which is the basis for our current search, is that you'd need to hack around it quite a lot. As mentioned in https://github.com/squidfunk/mkdocs-material/issues/6307#issuecomment-2079119984, facetted search is definitely coming to Material for MkDocs. Here's my current prototype that shows how search can be scoped to the top-level sections:

https://github.com/squidfunk/mkdocs-material/assets/932156/b8f1f4da-be78-4420-a5a9-0e4623bec617

Authors will be able to define arbitrary metadata for facetting search, e.g. categories, authors, tags, etc. Note that the UI is only for prototyping, it is not the final UI/UX design, so please be patient while I'm finishing it. It's quite some work.

> Is this possible - is there a "process_search_result" hook or something?

Nope, currently not.

**Reply by NFanoe** [reply]
Thanks for you reply! I guess I'll just wait for the new search :) 
(By the way, I can't see the video you posted, seems to be 0 seconds long or something.)

**Reply by NFanoe** [reply]
Oh, there it is, the video - just had to wait a while! :)
